lesson6/README.md

```diff
@@ -0,0 +1,87 @@
+
+#### 3. Testing after developing:
+Once you have a Minimum Viable Product (MVP) of your app, test again. But this time, it is a little different.
+  1. Try to break it
+      * Rotate the device. Do all elements adjust well?
+      * Tap at random places
+      * Try devices of different screen sizes.
+  2. Get a fresh pair of eyes
+  3. Test your code for edge cases and weird events 
+
+## App Elements
+Note: The documentation for all the app elements listed below are really good. We tried to summarize it so that you get an idea of each of the elements. When you use them in your app, the developers website is the best place to learn more about them.
+
+#### Layouts
+While writing the XML files for your app, you might have noticed there are numerous ways to layout and display the content. [This](https://developer.android.com/guide/topics/ui/declaring-layout.html) document explains layouts really well and talks about margins, padding, . Go read it!
+
+The two common layouts used in XML are:
+  1. [Linear Layout](https://developer.android.com/guide/topics/ui/layout/linear.html)
+The elements in a linear layout are stacked one after another, either horizontally or vertically. You get to specify the orientation. 
+
+  2. [Relative Layout](https://developer.android.com/guide/topics/ui/layout/relative.html)
+As the name suggests, all the elements are displayed relative to either other children or the parent view. It is more powerful as it eliminates the need for nested Linear Layouts.
+
+#### [App bar](https://developer.android.com/training/appbar/index.html)
+![AppBar](https://developer.android.com/images/training/appbar/appbar_sheets.png)
+
+Almost every app you use will have an app bar because it provides a visual structure and interactive elements that are familiar to users. The key functions of the app bar are as follows:
+  * A dedicated space for giving your app an identity and indicating the user's location in the app.
+  * Access to important actions in a predictable way, such as search.
+  * Support for navigation and view switching (with tabs or drop-down lists).
+
+#### [Settings](https://developer.android.com/guide/topics/ui/settings.html)
+Remember how one of the design principles said "Let me make it mine"? We want to provide them with good defaults but always provide easy ways to change those defaults. Settings page just does that.
+
+If you want to learn about how to design a Settings page, [read this guide](https://material.google.com/patterns/settings.html#settings-usage).
+
+#### [Pop-up messages](https://developer.android.com/training/snackbar/index.html)
+Whenever you need to provide the users with a prominent enough message, use pop-up messages. Most of the time, the user doesn't have to respond to the message. Also, the message disappears after a short while. The pop-up message should not block the users from using the app.
```

Comment:
There is a picture in the readme that shows the non-intrusive pop-up. Do we need to show anything more than that?